core/src/main/java/org/hibernate/ogm/massindex/impl/BatchIndexingWorkspace.java

```diff
@@ -66,18 +66,15 @@ public BatchIndexingWorkspace(GridDialect gridDialect, SearchIntegrator search,
 		this.monitor = monitor;
 	}
 
-	private EntityKeyMetadata getEntityKeyMetadata() {
-		OgmEntityPersister persister = (OgmEntityPersister) sessionFactory.getEntityPersister( indexedType.getName() );
-		return new DefaultEntityKeyMetadata( persister.getTableName(), persister.getRootTableIdentifierColumnNames() );
-	}
-
 	@Override
 	public void run() {
 		ErrorHandler errorHandler = searchIntegrator.getErrorHandler();
 		try {
-			final EntityKeyMetadata keyMetadata = getEntityKeyMetadata();
+			OgmEntityPersister persister = (OgmEntityPersister) sessionFactory.getEntityPersister( indexedType.getName() );
+			final EntityKeyMetadata keyMetadata = new DefaultEntityKeyMetadata( persister.getTableName(), persister.getRootTableIdentifierColumnNames() );
```

Comment:
I'm wondering why we don't re-use the EKM from the persister. I realize it's been like that before your change, but it seems not needed.